### PR TITLE
feat(ai): correlate alerts into incidents

### PR DESCRIPTION
**Pull request by shahargl**
close https://github.com/keephq/keep/issues/2331
close https://github.com/keephq/keep/issues/2332

### CODE REVIEW

File: keep/api/bl/ai_suggestion_bl.py [hunk 1]
```diff
@@ -0,0 +1,499 @@
...
+from keep.api.models.db.topology import TopologyServiceDtoOut
+
+
+class AISuggestionBl:
+    def __init__(self, tenant_id: str, session: Session | None = None) -> None:
+        self.logger = logging.getLogger(__name__)
+        self.tenant_id = tenant_id
+        self.session = session if session else get_session_sync()
+
+        # Todo: interface it with any model
```
Comment (VladimirFilonov) [inline keep/api/bl/ai_suggestion_bl.py:31]: Can we add issues and mention links for TODOs?

File: keep/api/bl/ai_suggestion_bl.py [hunk 2]
```diff
@@ -0,0 +1,499 @@
...
+        )
+
+        topology_text = "\n".join(
+            [
+                f"Topology {idx+1}: {json.dumps(topology.dict(), default=str)}"
+                for idx, topology in enumerate(topology_data)
+            ]
+        )
+
+        system_prompt = """
```
Comment (VladimirFilonov) [inline keep/api/bl/ai_suggestion_bl.py:363]: Possible future improvement: move prompt templates to the database for easier editing

File: keep/api/bl/incidents_bl.py
Comment (VladimirFilonov) [inline keep/api/bl/incidents_bl.py:1]: :heart:

File: keep/api/models/db/ai_suggestion.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    )
+
+    suggestion: AISuggestion = Relationship(back_populates="feedbacks")
+
+    class Config:
+        arbitrary_types_allowed = True
+
+
+"""
+SQL commands to create the tables in SQLite:
```
Comment (VladimirFilonov) [inline keep/api/models/db/ai_suggestion.py:57]: Why?

File: keep/api/bl/ai_suggestion_bl.py (hunk 1 above)
Comment (shahargl) [reply keep/api/bl/ai_suggestion_bl.py:31]: done

File: keep-ui/app/alerts/alert-create-incident-ai-modal.tsx [hunk 4]
```diff
@@ -0,0 +1,365 @@
...
+          },
+          body: JSON.stringify(
+            alertsToProcess.map((alert) => alert.fingerprint)
+          ),
+          signal: controller.signal,
+        });
+
+        // If timeout error (which happens after 30s with NextJS), wait 10s and retry
+        // This handles cases where the request goes through the NextJS server which has a 30s timeout
+        if (!response.ok && response.status === 500) {
```
[the remark this answers is not in the document]
Comment (shahargl) [reply keep-ui/app/alerts/alert-create-incident-ai-modal.tsx:95]: I didn't find way of doing that but I've added a TODO there with a ticket

**Review by talboren — APPROVED** [review]
LGTM!

**Pull request merged**